### ISSUE TITLE
Missing space in time format for 'fr-CA' on Linux

### ISSUE THREAD

**Issue by mbp** [dotnet/runtime#1340]
The following code:
```
var culture = CultureInfo.GetCultureInfo("fr-CA");
Console.WriteLine($"ShortDatePattern: {culture.DateTimeFormat.ShortDatePattern}");
Console.WriteLine($"ShortTimePattern: {culture.DateTimeFormat.ShortTimePattern}");
Console.WriteLine($"LongDatePattern: {culture.DateTimeFormat.LongDatePattern}");
Console.WriteLine($"LongTimePattern: {culture.DateTimeFormat.LongTimePattern}");
Console.WriteLine(DateTime.Now.ToString(culture));
```

Outputs on Linux/Docker .NET Core 3.1:

```
ShortDatePattern: yyyy-MM-dd
ShortTimePattern: HH 'h'mm
LongDatePattern: dddd d MMMM yyyy
LongTimePattern: HH 'h'mm 'min'ss 's'
2020-01-06 14 h37 min01 s
```

There is a space missing after "h" and "min", the last line should read:
```
2020-01-06 14 h 37 min 01 s
```

According to https://www.unicode.org/cldr/charts/latest/by_type/date_&_time.gregorian.html the format is supposed to be `HH 'h' mm 'min' ss 's'`

This is closely related to dotnet/corefx#37557, I believe it was not correctly fixed.

**Comment by danmosemsft** [comment]
cc @tarekgh

**Comment by tarekgh** [comment]
This is a bug in the logic 

https://github.com/dotnet/runtime/blob/c1a51e2c538a4c520458190d9c4a1645fc375525/src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.Unix.cs#L266

when finding the closing single quote, we should decrement back the i index. I'll submit a fix soon for this issue.

**Comment by mathcantin** [comment]
Is it possible to put the fix in version 3.1?

**Comment by tarekgh** [comment]
@mathcantin 3.1 is LTS release which we are careful with the changes goes there. I am wondering, in your case, can you use the workaround? it should be simple. Like, override the DateTimeFormatInfo property (e.g. LongTimePattern) with the right one.
